pom.xml

```diff
@@ -268,7 +268,8 @@
           </execution>
         </executions>
         <configuration>
-          <scalaVersion>2.10.4</scalaVersion>
+          <scalaVersion>${scala.version}</scalaVersion>
+          <recompileMode>incremental</recompileMode>
```

Comment:
We'd removed incremental in the past. I'd need to dig through the commit history to find the exact reason, but it has something to do with causing transient build issues. The Jenkins build just failed with an issue related to such. Alternatively, we could do `-DrecompileMode=all` in `scripts/jenkins-test` as suggested by https://github.com/davidB/scala-maven-plugin/issues/185. I'm open to either. It'd be nice to have faster Scala compile times in the general case.